pkg/storage/snapshot/snapshot.go

```diff
@@ -707,6 +708,11 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "Operation complete"))
 		updateSnapshotCondition(vmSnapshotCpy, newReadyCondition(corev1.ConditionTrue, "Operation complete"))
 		updateSnapshotSnapshotableVolumes(vmSnapshotCpy, content)
+		// Update "kubevirt_vmsnapshot_timestamp_seconds" metric if the snapshot succeeded
+		if vmSnapshot.Status != nil && vmSnapshot.Status.Phase == snapshotv1.Succeeded {
```

Comment:
1. you dont need this if since at the point status is for sure not nil and phase is being changed to succeeded in line 707.
2. you should send vmSnapshotCpy to NewVMSnapshotCreated since that will be the updated version of the snapshot. If you dont check the cpy its possible to never update this metric since the "original" vmsnapshot variable doesnt get updated here with the succeeded phase and if the snapshot will not get reconciled again, which is very plausible, then the metric will never get updated.